src/com/aca/narekhakobyan/homework1/items/concepts/RecordableGadgetWithMemory.java

```diff
@@ -0,0 +1,36 @@
+package com.aca.narekhakobyan.homework1.items.concepts;
+
+import java.util.Objects;
+import java.util.Scanner;
+
+public abstract class RecordableGadgetWithMemory extends MemoryGadget {
+    private final Integer recDuration;
```

Comment:
all gadgets have recDuration property so I guess you can escalate this property to base model